code-challenges.js

```diff
@@ -29,20 +48,79 @@ var myString2 = "delta"
+
+
+
+
 // --------------------4) Create a statement that evaluates two numbers and subtracts the smaller number from the larger number using the two sets of test variables provided below. Expected output: 15, 12
 
 var number1 = 42
 var number2 = 27
 
-// var number1 = 7
-// var number2 = 19
+if (number1 > number2){
+    console.log(number1 - number2)
+} else if (number2 > number1)
+    console.log(number2 - number1)
+
+// Created an if/else if statement that would determine if one number was greater than another. If the variable number1 is greater then console.log the value of number1 minus number2. Created the else if statement in case there are instances where number2 is the greater number then the function will still perform the operation of subtracting the greater number. I performed the main operation and then changed number2 variable to be the greater number to ensure that the function would subtract if number2 was the greater value
+
+
+var number1 = 7
+var number2 = 19
 
+if (number1 > number2){
+    console.log(number1 - number2)
+} else if (number2 > number1)
+    console.log(number2 - number1)
+// Created an if/else statement that would determine if one number was greater than another. If the variable number1 is greater then console.log the value of number1 minus number2. Created the else if statement in case there are instances where number2 is the greater number, then the function will still perform the operation of subtracting the greater number. I performed the main operation and then changed number2 variable to be the greater number to ensure that the function would subtract correctly if number2 was the greater value
 
 
 // --------------------5) Copy/paste your code from #4. Refactor your code to also accept non-number edge cases. Inform your user if the variable is not a number using the two sets of test variables provided below. Expected output: "Your input is not a number", 3
 
 var numberRefactor1 = 42
 var numberRefactor2 = "hello"
 
-// var numberRefactor1 = 27
-// var numberRefactor2 = 24
+if (typeof numberRefactor1 === "string"){
+    console.log("Your input is not a number")
+} else if (typeof numberRefactor2 === "string"){
+    console.log("Your input is not a number")
+// Step 1: Set up an if/else if statement that will recognize a string for each variable using typeof
+} else if (numberRefactor1 > numberRefactor2){
+    console.log(numberRefactor1 - numberRefactor2)
+} else if (numberRefactor2 > numberRefactor1)
+    console.log(numberRefractor2 - numberRefractor1)
+//Step 2: Adjust the code from problem 4 to reflect the new variables. My current console.log gives the output "Your input is not a number"
+
+    
+var numberRefactor1 = 27
+var numberRefactor2 = 24
+
+if (typeof numberRefactor1 === "string"){
+    console.log("Your input is not a number")
+} else if (typeof numberRefactor2 === "string"){
```

Comment:
This is all great and I appreciate you checking the type. I want to encourage you to think about all data types not just strings and numbers. 

we can do this by saying something like 
```javascript
if (typeof numberRefactor1 !== "number"){
    console.log("Your input is not a number")
```

that way the logic looks just like the pseudo code